com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -57,49 +57,98 @@ internal struct AgentAction
+    ///   providing the appropriate action.
     ///
-    /// At any step, an agent may be considered done due to a variety of reasons:
-    ///     - The agent reached an end state within its environment.
-    ///     - The agent reached the maximum # of steps (i.e. timed out).
-    ///     - The academy reached the maximum # of steps (forced agent to be done).
+    /// To trigger an agent decision automatically, you can attach a <see cref="DecisionRequester"/>
+    /// component to the Agent game object. You can also call the agent's <see cref="RequestDecision"/>
+    /// function manually. You only need to call <see cref="RequestDecision"/> when the agent is
+    /// in a position to act upon the decision. In many cases, this will be every [FixedUpdate]
+    /// callback, but could be less frequent. For example, an agent that hops around its environment
+    /// can only take an action when it touches the ground, so several frames might elapse between
+    /// one decision and the need for the next.
     ///
-    /// Here, an agent reaches an end state if it completes its task successfully
-    /// or somehow fails along the way. In the case where an agent is done before
-    /// the academy, it either resets and restarts, or just lingers until the
-    /// academy is done.
+    /// Use the <see cref="OnActionReceived"/> function to implement the actions your agent can take,
+    /// such as moving to reach a goal or interacting with its environment. Both
+    /// <see cref="CollectObservations"/> and <see cref="OnActionReceived"/> are called during the Unity
+    /// [FixedUpdate] phase.
+    /// 
+    /// When you call <see cref="EndEpisode"/> on an agent or the agent reaches its <see cref="maxStep"/> count,
+    /// its current episode ends. You can reset the agent -- or remove it from the
+    /// environment -- by implementing the <see cref="OnEpisodeBegin"/> function. An agent also
+    /// becomes done when the <see cref="Academy"/> resets the environment, which only happens when
+    /// the <see cref="Academy"/> receives a reset signal from an external process via the
+    /// <see cref="Academy.Communicator"/>.
+    /// 
+    /// The Agent class extends the Unity [MonoBehaviour] class. You can implement the
+    /// standard [MonoBehaviour] functions as needed for your agent. Since an agent's
+    /// observations and actions take place during the [FixedUpdate] phase, you should
+    /// only use the [MonoBehaviour.Update] function for cosmetic purposes. If you override the [MonoBehaviour]
+    /// methods, [OnEnable()] or [OnDisable()], always call the base Agent class implementations.
+    /// 
+    /// You can implement the <see cref="Heuristic"/> function to specify agent actions using
+    /// your own heuristic algorithm. Implementing a heuristic function can be useful
+    /// for debugging. For example, you can use keyboard input to select agent actions in
+    /// order to manually control an agent's behavior. 
```

Comment:
```suggestion
    /// order to manually control an Agent's behavior. 
```

Reply:
Not sure what is the rule. I though we always capitalize Agent but I might be wrong